### DISCUSSION TITLE
Using a local GeoTIFF image as terrain model

### DISCUSSION THREAD

**Discussion by bobfortin64** [visgl/deck.gl#7816] [General]
I am trying to use a local geotiff file with encoded rgb values  as the  elevation data in a Terrain layer.

I am not able to get it to load. Should this be  possible?
I tried specifying a loader but not sure that is corret.
I even tried to pass in the data as a base64 encoded string with the same result.

new deck.TerrainLayer({
id: "fe069e7b-7546-4a4a-b665-7e0e512679d2",
elevationData: "C:/Users/Bob.Fortin/Documents/SeattleElevation_LL_MRR.tif",
elevationDecoder: {"rScaler": 256, "gScaler": 1, "bScaler": 1/256, "offset": -32768},
wireframe: false,
visible: true,
bounds: [-123.74647,46.451224000000003,-120.96088,48.308283000000003],
pickable: true,
    operation: "terrain+draw",
    loaders: [GeoTIFFLoader]
}),
]

Thanks for any ideas.

-Bob

**Comment by ibgreen** [comment]
No this won't be possible. 
- Firstly, a browser can't access local files by path, only by file selection or drag and drop. 
- Secondly the terrain source needs to be served as tiles. 
- Thirdly, the loaders.gl GeoTIFFLoader is experimental and unfinished.

**Reply by bobfortin64** [reply]
Thanks for the reply. The third point might be the issue. 
I am already able to access local files by starting up the browser with certain options.
The example on https://deck.gl/docs/api-reference/geo-layers/terrain-layer shows how to use a single image as a mesh instead of tiles and I can make this work with .png image.

**Reply by ibgreen** [reply]
Hmm, I believe that you are right in that the terrain layer can use a single image, however the TerrainExtension that underpins the `terrain+draw` operation for arbitrary layers is based on tiled rendering behind the scenes.

**Comment by gnimmelf** [comment]
Hi guys,

I just went down the same path, o I am wondering if it is feasible in the foreseeable future that using a single as file, or a data-set, as `elevationData` on this will work?  

Use case is a pipeline that uses national geodata from downloadable sets per muncipiality and produces a 3d 'floating island' of that geographical area. The data-sets in this case are multiple geojson files and a single geotiff for elevation.

Cheers,
Flemming
